README.md

```diff
@@ -79,7 +79,97 @@ java -javaagent:path/to/opentelemetry-javaagent-all.jar \
      -jar myapp.jar
 ```
 
-### Configuration parameters (subject to change!)
+### Manually instrumenting
+
+> :warning: starting with 0.6.0, and prior to version 1.0.0, `opentelemetry-javaagent-all.jar`
+only supports manual instrumentation using the `opentelemetry-api` version with the same version
+number as the Java agent you are using. Starting with 1.0.0, the Java agent will start supporting
+multiple (1.0.0+) versions of `opentelemetry-api`.
+
+You'll need to add a dependency on the `opentelemetry-api` library to get started; if you intend to
+use the `@WithSpan` annotation, also include the `opentelemetry-extension-annotations` dependency.
+
+#### Maven
+
+```xml
+  <dependencies>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-api</artifactId>
+      <version>0.11.0</version>
+    </dependency>
+    <dependency>
+      <groupId>io.opentelemetry</groupId>
+      <artifactId>opentelemetry-extension-annotations</artifactId>
+      <version>0.11.0</version>
+    </dependency>
+  </dependencies>
+```
+
+#### Gradle
+
+```groovy
+dependencies {
+    compile('io.opentelemetry:opentelemetry-api:0.11.0')
+    compile('io.opentelemetry:opentelemetry-extension-annotations:0.11.0')
+}
+```
+
+#### Adding attributes to the current span
+
+A common need when instrumenting an application is to capture additional application-specific or
+business-specific information as additional attributes to an existing span from the automatic
+instrumentation. Grab the current span with `Span.current()` and use of the `setAttribute()`
+methods:
+
+```java
+import io.opentelemetry.api.trace.Span;
+
+// ...
+
+Span span = Span.current();
+span.setAttribute(..., ...);
+```
+
+#### Creating spans around methods with `@WithSpan`
+
+Another common situation is to capture a span around an existing first-party code method. The
+`@WithSpan` annotation makes this straightforward:
+
+```java
+import io.opentelemetry.extension.annotations.WithSpan;
+
+public class MyClass {
+  @WithSpan
+  public void MyLogic() {
+      <...>
+  }
+}
+```
+
+Each time the application invokes the annotated method, it creates a span that denote its duration
+and provides any thrown exceptions. Unless specified as an argument to the annotation, the span name
+will be `<className>.<methodName>`.
+
+##### Suppressing `@WithSpan` instrumentation
+
+Suppressing `@WithSpan` is useful if you have code that is over-instrumented using `@WithSpan`
+and you want to suppress some of them without modifying the code.
```

Comment:
I know it's copied, but while we're updating the docs, for this, I'm not quite sure how this happens. `@WithSpan` is quite an explicit addition to code, how could it lead to over-instrumentation? @trask do you know the background of this quote? Would be nice if there's a real-world example here

Reply:
I do know the background! see https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/344#issuecomment-620818687

I don't think this section needs to be in top-level README, though, probably should be moved to https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/docs/suppressing-instrumentation.md

Reply:
I would imagine this happens if you're pulling in a dependency that uses `@WithSpan` and you need to suppress _their_ usage.

Reply:
I think `OTEL_JAVAAGENT_EXCLUDE_CLASSES` is supposed to work for that case pretty well. So anyways, maybe I'll propose removing the flag but we don't have to do anything here :)